### PR TITLE
[docs-only] Development guide && repo conventions

### PR DESCRIPTION
**Pull request by pascalwengerter**
## Description
Please feel free to push to this branch and/or comment your personal notes

## Motivation and Context
Makes code review easier and helps (new) contributors

### CODE REVIEW

**Comment by pascalwengerter** [comment]
## Naming of files/items/resources

*Codebase:* Files & folders get treated as "resources"

*User facing:* Files & folders, if both are involved "items" or (also) files. Shares if it's not owned by the current user

**Comment by pascalwengerter** [comment]
Make sure all date handling inside the project is done via `ISO` format

**Comment by pascalwengerter** [comment]
Try to avoid `.then` in favor of async/await

File: docs/development.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+This is a collection of tips and conventions to follow when working on [the ownCloud web frontend](https://github.com/owncloud/web). 
+It is a living document, please open a PR if you find something missing.
+
+## Contributing to ownCloud web
+
+Everyone is invited to contribute. Simply fork [the codebase](), check [the issues](https://github.com/owncloud/web/issues?q=is%3Aopen+is%3Aissue+label%3ATopic%3Agood-first-issue) for a suitable one and open a pull request!
+
+### Linting and tests
+
+To make sure your pull request can be efficiently reviewed and won't need a lot of changes down the road, please run the linter and the unit tests via `yarn lint --fix` and `yarn test:unit` locally. [Our CI](https://drone.owncloud.com/owncloud/web) will run on pull requests and report back any problems after that.
```
Comment (fschade) [inline docs/development.md:21]: event if we're not sure how to proceed with the integration tests... but i think right now we still have to mention those. Thoughts!? Also can we link the testing page here?

File: docs/development.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+### Changelog items
+
+In our project, we follow [SemVer](https://semver.org/) and keep a changelog for every change that influences the user experience (where "users" can be admins, end-users and extension developers).
+Some changes, like refactoring, updating dependencies or adding tests don't require a changelog item.
+
+Please add a changelog item to the `changelog/unreleased/` folder, referencing the issue and pull request numbers.
```
Comment (fschade) [inline docs/development.md:28]: can you mention the [template](https://github.com/owncloud/web/blob/master/changelog/TEMPLATE)

File: docs/development.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+## Code Conventions
```
Comment (fschade) [inline docs/development.md:30]: I LOVE THAT ❤️

**Review by fschade — APPROVED** [review]
just a little thing (optional, just a brain dump) 🎸

**Comment by pascalwengerter** [comment]
Ok let me follow up on your suggestions later and then hit that sweet merge btn!

File: docs/development.md (hunk 1 above)
Comment (pascalwengerter) [reply docs/development.md:21]: Adding integration tests & the link to the testing page, great suggestion

**Pull request merged**